pkg/registry/core/service/strategy_test.go

```diff
@@ -229,6 +232,257 @@ func TestDropDisabledField(t *testing.T) {
 
 }
 
+func TestDropServiceStatusDisabledFields(t *testing.T) {
+	ipModeVIP := api.LoadBalancerIPModeVIP
+	ipModeProxy := api.LoadBalancerIPModeProxy
+
+	testCases := []struct {
+		name          string
+		ipModeEnabled bool
+		svc           *api.Service
+		oldSvc        *api.Service
+		compareSvc    *api.Service
+	}{
+		/*LoadBalancerIPMode disabled*/
+		{
+			name:          "LoadBalancerIPMode disabled, ipMode not used in old, not used in new",
+			ipModeEnabled: false,
+			svc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{Ingress: []api.LoadBalancerIngress{
+					{
+						IP: "1.2.3.4",
+					},
+				}}
+			}),
+			oldSvc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{}
+			}),
+			compareSvc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{Ingress: []api.LoadBalancerIngress{
+					{
+						IP: "1.2.3.4",
+					},
+				}}
+			}),
+		},
+		{
+			name:          "LoadBalancerIPMode disabled, ipMode used in old and in new",
+			ipModeEnabled: false,
+			svc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{Ingress: []api.LoadBalancerIngress{
+					{
+						IP:     "1.2.3.4",
+						IPMode: &ipModeProxy,
+					},
+				}}
+			}),
+			oldSvc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{Ingress: []api.LoadBalancerIngress{
+					{
+						IP:     "1.2.3.4",
+						IPMode: &ipModeVIP,
+					},
+				}}
+			}),
+			compareSvc: makeValidServiceCustom(func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{Ingress: []api.LoadBalancerIngress{
+					{
+						IP:     "1.2.3.4",
+						IPMode: &ipModeProxy,
```

Comment:
Do I read this right?  We will NOT accept the update?

Reply:
In this case, we accepted the update. The value was updated from "ipModeVIP" to "ipModeProxy".